### PR TITLE
Mouseposition

### LINKED ISSUE #1296: Filtre for coordinates
To display coordinates with a custom template. For DMS and for numerical coordinates.

(Will be used in infobar, contextual info, redlining, etc.)

### PR DESCRIPTION
**Pull request by ger-benjamin**
Revival of #1124 
Fix #1296
Part of infobar. Specs here:  https://github.com/camptocamp/c2cgeoportal/wiki/Spec-%231668-Map-(Info-bar)

Example: http://ger-benjamin.github.io/ngeo/mouseposition/examples/contribs/gmf/mouseposition.html
Doc (search `xxxCoordnates` under `filters`): http://ger-benjamin.github.io/ngeo/mouseposition/apidoc/

### CODE REVIEW

File: src/services/filters.js [hunk 1]
```diff
@@ -21,11 +21,164 @@ goog.require('ngeo');
...
+ * Format a couple of numbers as Swiss coordinates.
+ *
+ * Example without parameters:
+ *
+ *      <p>{{[600000.1234, 200000.5678] | ngeoSwissCoordinates}}</p>
+ *      <!-- will Become 600'000, 200'001 -->
+ *      <p>{{[2600000.1234, 1200000.5678] | ngeoSwissCoordinates}}</p>
+ *      <!-- will Become 2'600'000, 1'200'001 -->
+ *
+ * Example with defined fractionDigits, prefix, separator and suffix (you can
```
Comment (fredj) [inline src/services/filters.js]: `fractionDigits` is not used here (not available)

**Comment by fredj** [comment]
`goog.require('ol.coordinate');` could be added in `src/services/filters.js`

**Comment by ger-benjamin** [comment]
> goog.require('ol.coordinate'); could be added in src/services/filters.js

Done

File: contribs/gmf/examples/mouseposition.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+goog.provide('gmf-mouseposition');
+
+/** @suppress {extraRequire} */
+goog.require('gmf.mapDirective');
+goog.require('gmf.mousepositionDirective');
+goog.require('ngeo.proj.EPSG2056');
+goog.require('ngeo.proj.EPSG21781');
+goog.require('ngeo.proj.EPSG32631');
```
Comment (fredj) [inline contribs/gmf/examples/mouseposition.js]: `EPSG32631` is not used, can this `goog.require` be removed ?

File: src/services/filters.js [hunk 3]
```diff
@@ -21,11 +22,164 @@ goog.require('ngeo');
...
+   * @param {ol.Coordinate} coordinates Array of two numbers.
+   * @param {number=} opt_fractionDigits number of digit, Same as
+   *     {@link ol.coordinate.format} opt_fractionDigits parameters.
+   * @param {string=} opt_prefix Optional prefix.
+   * @param {string=} opt_separator Optional separatator. default to ' '.
+   * @param {string=} opt_suffix Optional suffix.
+   * @return {string} East north formated coordinates.
+   */
+  var filterFn = function(coordinates, opt_fractionDigits, opt_prefix,
+      opt_separator, opt_suffix) {
```
Comment (sbrunner) [inline src/services/filters.js]: Instance og havine a `opt_prefix`, `opt_separator`, `opt_suffix` can't we have a template and used for example https://www.npmjs.com/package/es6-template-strings

File: src/services/filters.js [hunk 4]
```diff
@@ -21,11 +22,164 @@ goog.require('ngeo');
...
+ *      <p>{{[2600000.1234, 1200000.5678] | ngeoSwissCoordinates:[:; :]}}</p>
+ *      <!-- will Become [2'600'000; 1'200'001] -->
+ *
+ * @return {function(ol.Coordinate, string, string, string): string} A
+ *     function to format numbers into an Swiss coordinates string.
+ * @ngInject
+ * @ngdoc filter
+ * @ngname ngeoSwissCoordinates
+ */
+ngeo.SwissCoordinates = function() {
```
Comment (sbrunner) [inline src/services/filters.js]: Do we relay need this? for me we can do every thing with EastNorthCoordinates with a format...

Comment (sbrunner) [inline src/services/filters.js]: Or with a replace as in ol3: https://github.com/openlayers/ol3/blob/master/src/ol/coordinate.js#L149

**Comment by ger-benjamin** [comment]
@sbrunner I've added a template-like systeme and totally remove one of my filters (there remainder on for DMS and one for "number" coordinates)

File: src/services/filters.js [hunk 5]
```diff
@@ -21,11 +21,140 @@ goog.require('ngeo');
...
+   *     "." as the decimal separator and include "'" group separators after
+   *     each third digit).
+   * @return {string} East north formated coordinates.
+   */
+  var filterFn = function(coordinates, opt_fractionDigits, opt_template,
+      opt_localize) {
+    var template = opt_template ? opt_template : '{x} {y}';
+    var x = coordinates[0];
+    var y = coordinates[1];
+    if (opt_localize) {
```
Comment (sbrunner) [inline src/services/filters.js]: => `if (opt_localize !== false) {` ?
I think that's should be true by default...

**Comment by sbrunner** [comment]
Just a small remark then that's looks good :-)

**Comment by ger-benjamin** [comment]
Done. Thanks @sbrunner , @fredj

**Comment by ger-benjamin** [comment]
Can I merge ?

**Comment by fredj** [comment]
+1

**Comment by ger-benjamin** [comment]
Okay, (I've seen typo in the doc, I'll correct that then merge).

**Pull request merged**